### ISSUE TITLE
More than one player picking up the same item when opening the chest (Container) when they click to pick the item at the same time

### ISSUE THREAD

**Issue by LucianoMonsores** [kamcknig/pick-up-stix#102]
**Describe the bug**
If more than one player opens a chest (Container) at the same time, they can take the same item if they click to pick the item at the same time.

The item is removed if only one player picks it up and closes the chest (Container) or if the item has enough time to be removed. I don't know if it is possible to lock the first item clicked. This would prevent other players from picking up the item improperly, perhaps due to a delay in removing the item.

**To Reproduce**
1. Access the game using 2 player accounts or more.
2. Opens the chest (Container) logged in 2 different player accounts.
3. Click to pick up an item with quantity 1 with both players at the same time.
4. You can see in the chat that more than one player picked up the same item at the same time.

**Expected behavior**
The first item should have a lock for the first player to click to pick it up from the chest.

**Screenshots**
![image](https://user-images.githubusercontent.com/26168222/91635828-938c6d80-e9f3-11ea-8de0-424243925fb9.png)

**Foundry Version**
0.6.5

**Pick-Up-Stix Version**
0.8.9

**Comment by kamcknig** [comment]
@LucianoMonsores When you say "at the same time," do you actually mean as close to at the same time as possible? I want to make sure this is actually a race condition. Because I do have code that should update the quantity on all player's screens. If this is a race condition, it seems like it would be hard to do it accidentally.

**Comment by LucianoMonsores** [comment]
Hi @kamcknig, thanks for getting back to me.

Yes, as close as possible at the same time. This happened last Friday with me, 3 players opened a chest (It was the first time we had this in the game) and there was a single item and the players practically clicked at the same time and I believe that there was not enough time for the item to be removed.

I know it is something difficult to solve, I suggested Lock's solution but I don't know if this is possible to be implemented for Foundry. I see this a lot in the shopping cart systems. 

The way I found it as a workaround solution, was to deselect the option to close the chest so that only one player can get the item and then he can drag and drop the item to the floor or to another token.
